src/ast/ast_rewrite_star_projections.c

```diff
@@ -170,6 +170,15 @@ static void replace_clause
 ) {
 	cypher_astnode_type_t t = cypher_astnode_type(clause);
 
+	//--------------------------------------------------------------------------
+	// return early in the case of a parse error
+	//--------------------------------------------------------------------------
+	uint child_count = cypher_astnode_nchildren(root);
+	for(uint i = 0; i < child_count; i++) {
+		const cypher_astnode_t *to = cypher_astnode_get_child(root, i);
+		if(cypher_astnode_type(to) == CYPHER_AST_ERROR) return;
+	}
+
```

Comment:
I think it would be better to use `AST_ContainsErrors`
After this check:`if(!cypher_parse_result_eof(result)) {...` and before invoking `AST_RewriteStarProjections`